rclpy/test/test_parameter.py

```diff
@@ -259,23 +259,77 @@ def test_parameter_value_to_python(self):
-        yaml_string = """/param_test_target:
-            ros__parameters:
-                param_1: 1
-                param_str: string
+        yaml_string = """
+            /param_test_target:
+                ros__parameters:
+                    abs-nodename: true
+            param_test_target:
+                ros__parameters:
+                    base-nodename: true
+            /foo/param_test_target:
+                ros__parameters:
+                    abs-ns-nodename: true
+            /foo:
+                param_test_target:
+                    ros__parameters:
+                        abs-ns-base-nodename: true
+            /bar/param_test_target:
+                ros__parameters:
+                    abs-ns-nodename: false
+            /bar:
+                param_test_target:
+                    ros__parameters:
+                        abs-ns-base-nodename: false
+            /**:
+                ros__parameters:
+                    wildcard: true
             """
-        expected = {
-            'param_1': Parameter('param_1', Parameter.Type.INTEGER, 1).to_parameter_msg(),
-            'param_str': Parameter('param_str', Parameter.Type.STRING, 'string').to_parameter_msg()
+
+        # target nodes is specified, so it should only parse wildcard
+        expected_no_target_node = {
+            'wildcard': Parameter('wildcard', Parameter.Type.BOOL, True).to_parameter_msg(),
+        }
+        # target nodes is specified with wildcard enabled
+        expected_target_node_wildcard = {
+            'wildcard': Parameter(
+                'wildcard', Parameter.Type.BOOL, True).to_parameter_msg(),
+            'abs-nodename': Parameter(
+                'abs-nodename', Parameter.Type.BOOL, True).to_parameter_msg(),
+            'base-nodename': Parameter(
+                'base-nodename', Parameter.Type.BOOL, True).to_parameter_msg(),
+        }
+        # target nodes is specified with wildcard disabled
+        expected_target_node = {
+            'abs-nodename': Parameter(
+                'abs-nodename', Parameter.Type.BOOL, True).to_parameter_msg(),
+            'base-nodename': Parameter(
+                'base-nodename', Parameter.Type.BOOL, True).to_parameter_msg(),
+        }
+        # target nodes is specified with wildcard and namespace
+        expected_target_node_ns = {
+            'wildcard': Parameter('wildcard', Parameter.Type.BOOL, True).to_parameter_msg(),
+            'abs-ns-nodename': Parameter(
+                'abs-ns-nodename', Parameter.Type.BOOL, True).to_parameter_msg(),
+            'abs-ns-base-nodename': Parameter(
+                'abs-ns-base-nodename', Parameter.Type.BOOL, True).to_parameter_msg(),
         }
 
         try:
             with NamedTemporaryFile(mode='w', delete=False) as f:
                 f.write(yaml_string)
                 f.flush()
                 f.close()
-                parameter_dict = parameter_dict_from_yaml_file(f.name)
```

Comment:
It seems that you changed the test case for using wildcard from `false` to `true` explicitly. Please notice that the API is exported and used in https://github.com/ros2/demos/blob/9c4ced3c5be392145312e0c0d3653140a2e29cc0/demo_nodes_py/demo_nodes_py/parameters/async_param_client.py#L76.

Reply:
good eye 👍 here is the fix https://github.com/ros2/demos/pull/656